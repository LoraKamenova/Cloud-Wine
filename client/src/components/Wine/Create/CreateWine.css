.add-wine-form-wrapper > .form-header-wrapper > .form-header-heading-container > .form-header-heading {
    left: 60px;
    bottom: -30px;
}
