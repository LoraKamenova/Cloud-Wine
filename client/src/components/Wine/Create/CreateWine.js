import './CreateWine.css'
import FormHeader from "../../Shared/FormHeader/FormHeader";
import InputError from "../../Shared/InputError/InputError";

import * as wineService from "../../../services/wineService";

import {useHistory} from "react-router-dom"
import {useState} from "react";

const CreateWine = () => {
    const history = useHistory();

    const [errorMessage, setErrorMessage] = useState('');

    const onCreateWineSubmitHandler = (e) => {
        e.preventDefault();

        const {
            name,
            category,
            description,
            producer,
            imageUrl1,
        } = e.target;

        wineService.create(
            name.value,
            category.value,
            description.value,
            producer.value,
            imageUrl1.value)
            .then(() => {
                history.push('/wine/all')
            })
    };

    const onChangeDescriptionHandler = (e) => {
        if (e.target.value.length < 100) {
            setErrorMessage('Описанието трябва да е минимум 100 символа');
        } else {
            setErrorMessage('');
        }
    };

    return (
        <section className="form-wrapper add-wine-form-wrapper">
            <FormHeader title="Ново вино"/>
            <form className="form-container" onSubmit={onCreateWineSubmitHandler}>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="name">Име:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="name"
                           name="name"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="category">Категория:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="category"
                           name="category"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="producer">Производител:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="producer"
                           name="producer"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="description">Описание:</label>
                    </div>
                    <textarea className="form-field-input textarea"
                              rows="11"
                              id="description"
                              name="description"
                              required
                              onBlur={onChangeDescriptionHandler}/>
                    <InputError>{errorMessage}</InputError>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="imageUrl1">URL снимка:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="imageUrl1"
                           name="imageUrl1"
                           required/>
                </div>

                <div className="form-button-wrapper">
                    <input className="form-button" type="submit" value="ИЗПРАТИ"/>
                </div>

            </form>
        </section>
    );
};

export default CreateWine;
