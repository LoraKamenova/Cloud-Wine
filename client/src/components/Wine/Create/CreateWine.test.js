import { render } from '@testing-library/react';
import CreateWine from './CreateWine';

describe('CreateWine Component', () => {

    it("CreateWine Component snapshot", () => {
        const { asFragment } = render(<CreateWine />);
        expect(asFragment()).toMatchSnapshot();
    });

});
