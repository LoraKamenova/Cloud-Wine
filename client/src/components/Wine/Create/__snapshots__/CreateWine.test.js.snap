// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWine Component CreateWine Component snapshot 1`] = `
<DocumentFragment>
  <section
    class="form-wrapper add-wine-form-wrapper"
  >
    <div
      class="form-header-wrapper"
    >
      <div
        class="form-header-heading-container"
      >
        <h1
          class="form-header-heading"
        >
          Ново вино
        </h1>
      </div>
      <div
        class="form-header-image-container"
      >
        <img
          alt="vine"
          class="form-header-image"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1615712293/React%20Project/grape_border_ogd8gq.png"
        />
      </div>
      <h1
        class="form-header-heading-hidden"
      >
        Ново вино
      </h1>
    </div>
    <form
      class="form-container"
    >
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="name"
          >
            Име:
          </label>
        </div>
        <input
          class="form-field-input"
          id="name"
          name="name"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="category"
          >
            Категория:
          </label>
        </div>
        <input
          class="form-field-input"
          id="category"
          name="category"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="producer"
          >
            Производител:
          </label>
        </div>
        <input
          class="form-field-input"
          id="producer"
          name="producer"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="description"
          >
            Описание:
          </label>
        </div>
        <textarea
          class="form-field-input textarea"
          id="description"
          name="description"
          required=""
          rows="11"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="imageUrl1"
          >
            URL снимка:
          </label>
        </div>
        <input
          class="form-field-input"
          id="imageUrl1"
          name="imageUrl1"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-button-wrapper"
      >
        <input
          class="form-button"
          type="submit"
          value="ИЗПРАТИ"
        />
      </div>
    </form>
  </section>
</DocumentFragment>
`;
